<?xml version="1.0"?>
<launch>
    
    
        <arg name="robotName" default="ropod"/>  
        <arg name="laser1_name" value ="laser"/>
        <!--<arg name="map_file" default="$(find ed_object_models)/models/hospital_test_amcl/walls/map_amcl.yaml" />-->
        <!--<arg name="map_file" default="$(find ropod_navigation_test)/config/amk-f064/map.yaml"/>-->
        <!--<arg name="map_file" default="/home/cesar/ropod-project-software/catkin_workspace/src/applications/occupancy_grids/tue/lab-level/map.yaml"/>-->
         <arg name="map_file" default="$(find ed_object_models)/models/tue_hallway_RLtoAL/walls/shape/map.yaml" />
       
        
   <!-- Run the map server -->
     <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
  
    <!--- Run AMCL -->
    <node pkg="amcl" type="amcl" name="amcl">
        <remap from="scan" to="/$(arg robotName)/$(arg laser1_name)/scan"/>

        <!--- Odometery model parameters  -->
        <param name="odom_model_type" value="omni"/>
        <param name="odom_alpha1" value="7.0"/>
        <param name="odom_alpha2" value="5.0"/>
        <param name="odom_alpha3" value="5.0"/>
        <param name="odom_alpha4" value="5.0"/>
        <param name="odom_alpha5" value="7.0"/>
        <param name="odom_frame_id" value="$(arg robotName)/odom"/>
        <param name="base_frame_id" value="$(arg robotName)/base_link"/>
        <param name="global_frame_id" value="/map"/>

        <!-- Overall filter parameters -->
        <param name="min_particles" value="200"/>
        <param name="max_particles" value="1000"/>
        <param name="kld_err" value="0.01"/>
        <param name="kld_z" value="0.99"/>
        <param name="update_min_d" value="0.05"/>
        <param name="update_min_a" value="0.01"/>
        <param name="resample_interval" value="2"/>
        <param name="transform_tolerance" value="0.1" />
        <param name="recovery_alpha_slow" value="0.0"/>
        <param name="recovery_alpha_fast" value="0.0"/>
        <param name="gui_publish_rate" value="-1"/>
        <param name="use_map_topic" value="true"/>
        <param name="first_map_only" value="true"/>

        <!--Laser model parameters -->
        <param name="laser_min_range" value="0.04"/>
        <param name="laser_max_range" value="5.0"/>
        <param name="laser_max_beams" value="30"/>
        <param name="laser_z_hit" value="0.95"/>
        <param name="laser_z_short" value="0.01"/>
        <param name="laser_z_max" value="0.05"/>
        <param name="laser_z_rand" value="0.05"/>
        <param name="laser_sigma_hit" value="0.2"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_likelihood_max_dist" value="1.0"/>
        <param name="laser_model_type" value="likelihood_field"/>
        
        <!-- initial pose of the robot -->
        <param name="initial_pose_x" value="3.1"/>
        <param name="initial_pose_y" value="15.3"/>
        <param name="initial_pose_a" value="-1.54"/>
        
    </node>
    
    <!-- Launch ED -->
    <node pkg="ed" type="ed" name="ed" args="$(find ropod_navigation_test)/config/model-example-ropod-navigation-ED.yaml">
        <remap from="~goto_action" to="/ropod_task_executor/GOTO" />
    </node>
    
    <!-- Launch navigation -->
    <node name="maneuver_navigation" pkg="maneuver_navigation" type="maneuver_navigation" output="screen">             
        <rosparam file="$(find ropod_navigation_test)/config/parameters/footprint_ropod.yaml" command="load" ns="local_costmap" />       
        <rosparam file="$(find ropod_navigation_test)/config/parameters/local_costmap_params_EDhallway.yaml"  command="load" ns="local_costmap" />    
        
<!--        <rosparam file="$(find ropod_navigation_test)/config/parameters/base_local_planner_params.yaml" command="load" />
    <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />        -->

        <rosparam file="$(find ropod_navigation_test)/config/parameters/teb_local_planner_params_ropod.yaml" command="load" />
        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />

        
        <remap from="/maneuver_navigation/cmd_vel" to="/load/cmd_vel"/>
        <remap from="odom" to="/load/odom"/>        
    </node>       
    
</launch>
